tests/python/unittest/test_higher_order_grad.py

```diff
@@ -15,13 +15,52 @@
 # specific language governing permissions and limitations
 # under the License.
 
-import math
 
+import math
 from mxnet import nd, autograd
 from mxnet.test_utils import assert_almost_equal, random_arrays
 from common import with_seed
 
 
+@with_seed()
+def test_sin():
+    def sin(x):
+        return nd.sin(x)
+
+    def grad_grad_op(x):
+        return -nd.sin(x)
+
+    arrays = random_arrays((2, 2), (2, 3), (4, 5, 2), (3, 1, 4, 5))
+    for array in arrays:
+        check_second_order_unary(array, sin, grad_grad_op)
```

Comment:
I think `check_second_order_unary` function should be moved to `python/mxnet/test_utils.py`

Reply:
It's only used in this test. If we add a different test file then it makes sense as you suggested.